values.yaml

```diff
@@ -85,41 +86,66 @@ st2:
       name: st2packs
       tag: latest
       pullPolicy: Always
-
-    # https://docs.stackstorm.com/reference/ha.html#st2sensorcontainer
-    # It is possible to run st2sensorcontainer in HA mode by running one process on each compute instance.
-    # Each sensor node needs to be provided with proper partition information to share work with other sensor
-    # nodes so that the same sensor does not run on different nodes.
-    sensors:
-      # Specify default container that executes all sensors.
-      # To partition sensors with one sensor per node, override st2.packs.sensors.
-      # NOTE: Do not modify this file.
-      - name:
-        livenessProbe: {}
-        readinessProbe: {}
-        # TODO: Find out recommended/default resources for this specific service (#5)
-        resources: {}
-        # Additional advanced settings to control pod/deployment placement
-        affinity: {}
-        nodeSelector: {}
-        tolerations: []
   # Import data into StackStorm's Key/Value datastore (https://docs.stackstorm.com/datastore.html)
   keyvalue:
     #- name: st2_version
     #  scope: st2kv.system
     #  secret: false
     #  encrypted: false
     #  value: "2.9"
-  # Import a list of ST2 API Keys (https://docs.stackstorm.com/authentication.html#api-key-migration)
-  apikeys:
-    #- created_at: '2018-12-15T00:21:48.507388Z'
-    #  enabled: true
-    #  id: 5c14491c6cb8de1a9207e3a2
-    #  key_hash: 56928c2d9637ce44338e9564d4b939df8b258410db23b5a80f8ad69d58e648b574f35f9293c3a76bde263738be9aa8379a81553cd55513ad672540b7b0ec0cac
-    #  metadata: {"comment": "Example unsecure ST2 API key from K8s HA Helm values.yaml"}
-    #  uid: api_key:56928c2d9637ce44338e9564d4b939df8b258410db23b5a80f8ad69d58e648b574f35f9293c3a76bde263738be9aa8379a81553cd55513ad672540b7b0ec0cac
-    #  user: st2admin
+  # Experimental st2chatops integration for StackStorm-HA Helm chart.  (https://github.com/mosn/stackstorm-ha/issues/17)
+  chatops:
+    gloal.yaml: |
+      ---
+      image: "stackstorm/st2chatops"
+      reoisitory:
+      hubot_name: hubot
+      hubot_alias: '!'
+      node_tls_reject_unauthorized: false
+      express_port: 8081
+      hubot_log_level: info
+      hubot_adapter: matteruser
+    st2.yaml: |
+      ---
+      ip: 
+      groups:
+      user:
+      password:
+      wss_port: 443
+      http_port: 443
+      reply: true
+      log_level: info
+      reply: true
+    mattermost.yaml: |
```

Comment:
Instead of having dedicated section for each possible hubot adapter (which could be outdated with time), would be better for user to specify any custom ENV variables they want to pass to `st2chatops` container.

Those custom ENVs will be applied on top of `st2chatops.env` like https://github.com/StackStorm/st2chatops/blob/c99d779d6e67385156bfd11b4c9997cf7e6f7ca8/st2chatops.env#L25-L26 in this example.

Reply:
Can you provide an example implementation of this?
the logic is that If a user is left with the responsibility of adding key/values for the custom adapter, then I must find a place to properly document the right values for it.  `values.yaml` is a candidate but then this is will make it crowded and filled with unrelated information.

Another way that I can think of is to separate custom values into its own config file and the user can reference the corresponding configuration file in values.yaml. This way we allow the same configuration file to be used by multiple applications and also we can have documentation for each if the custom values while keeping `values.yaml` nice and clean. 

We can use the same approach for all St2 components. A good implementation of this approach can be seen with   [Gitlan Providers.
](https://docs.gitlab.com/charts/charts/globals.html#providers)

Reply:
Left an example in https://github.com/StackStorm/stackstorm-ha/pull/55#pullrequestreview-218331459